src/mnt/fuse_pure.rs

```diff
@@ -23,9 +23,11 @@ use std::process::{Command, Stdio};
 use std::sync::Arc;
 use std::{mem, ptr};
 
-const FUSERMOUNT_BIN: &str = "fusermount";
-const FUSERMOUNT3_BIN: &str = "fusermount3";
-const FUSERMOUNT_COMM_ENV: &str = "_FUSE_COMMFD";
+#[cfg(target_os = "linux")]
+const FUSE_PROGRAM_NAME: &str = "fusermount";
+
+#[cfg(target_os = "macos")]
+const FUSE_PROGRAM_NAME: &str = "macfuse";
```

Comment:
I think this should be `mount_macfuse` based on the binary used in `detect_fuse_program`